go/store/nbs/table_set.go

```diff
@@ -286,7 +287,13 @@ func (ts tableSet) Size() int {
 
 // append adds a memTable to an existing tableSet, compacting |mt| and
 // returning a new tableSet with newly compacted table added.
-func (ts tableSet) append(ctx context.Context, mt *memTable, checker refCheck, stats *Stats) (tableSet, error) {
+func (ts tableSet) append(ctx context.Context, mt *memTable, checker refCheck, hasCache *lru.TwoQueueCache[addr, struct{}], stats *Stats) (tableSet, error) {
```

Comment:
tableSet is currently immutable (I think?) and we use the hasCache outside of it in the errorIfPending...I don't think I want the same hasCache hanging on multiple table sets because it would be nice if we could eventually remove some unnecessary locking overhead if it turns out to be a win...